faunadb-java/src/main/java/com/faunadb/client/query/Pagination.java

```diff
@@ -0,0 +1,175 @@
+package com.faunadb.client.query;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.faunadb.client.types.Value;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A pagination expression.
+ * <p>
+ * <b>Example:</b>
+ * We can paginate all instances of a class, with its evets, in pages with 4 elements only.
+ * <pre>{@code
+ * client.query(
+ *    Paginate(Match(Ref("indexes/all_spells"))
+ *      .withEvents(true)
+ *      .withSize(4)
+ * );
+ * }</pre>
+ *
+ * @see Language#Paginate(Expr)
+ * @see <a href="https://faunadb.com/documentation/queries#read_functions">FaunaDB Read Functions</a>
+ */
+public final class Pagination extends Fn.Call {
```

Comment:
It is slightly confusing that this inherits Fn.Call, but doesn't really use much of it. If this just inherited from Expr, and stored the resource in another field, it would be a bit clearer what's going on.